### ISSUE TITLE
[no-unused-vars-experimental] False positive for arguments in function override in derived class

### ISSUE THREAD

**Issue by lgarron** [typescript-eslint/typescript-eslint#1321]
**Repro**

```JSON
{
  "rules": {
    "no-unused-vars": [
      "off"
    ],
    "@typescript-eslint/no-unused-vars-experimental": [
      "error"
    ]
  }
}
```

```TS
class A {
  scalePoints(x: number): number {
    return 2 * x;
  }
}

class B extends A {
  scalePoints(x: number): number {
    /* Simplification of 0 * x. */
    return 0;
  }
}

const maybeB: A | B = new B();
console.log(maybeB.scalePoints(3));

const definitelyB: B = new B();
console.log(definitelyB.scalePoints(3));
```

**Expected Result**
No error.

**Actual Result**
The `x: number` argument to `B.scalePoints` is marked as an `no-unused-vars-experimental` error.

**Additional Info**

At the core here is that the implementation of a function override may "happen" not to depend on a certain argument at a given point in time, but may do so in general. It's important to me that I don't have to modify implementations or calls to subclasses based on whether the implementation of a given function happens to depend on the general argument for the method.

That is, removing the the argument from `B.scalePoints` will cause an error in the final line.  I would prefer that this line can keep the call as `definitelyB.scalePoints(3)`.

There are also so many cases in my project that I'd prefer not to disable the linter for each line. I would also prefer not to disable the rule for a whole file, because I still want to catch errors apart from this.

See https://github.com/cubing/cubing.js/blob/d0204f3d6173058fa18badb968e49af15d852083/src/twisty/cursor.ts#L16 for a real code example.

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.11.0` |
| `@typescript-eslint/parser`               | `2.11.0` |
| `TypeScript`                              | `3.5.2` |
| `ESLint`                                  | `6.7.2` |
| `node`                                    | `12.4.0` |
| `npm`                                     | `6.13.0` |

**Comment by bradzacher** [comment]
There is no false positive here.
The parameter is unused.

[You would get the exact same error were you using the base eslint `no-unused-vars`](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJlcnJvclwiXSAqL1xuXG5jbGFzcyBBIHtcbiAgc2NhbGVQb2ludHMoeCkge1xuICAgIHJldHVybiAyICogeDtcbiAgfVxufVxuXG5leHBvcnQgY2xhc3MgQiBleHRlbmRzIEEge1xuICBzY2FsZVBvaW50cyh4KSB7XG4gICAgLyogU2ltcGxpZmljYXRpb24gb2YgMCAqIHguICovXG4gICAgcmV0dXJuIDA7XG4gIH1cbn1cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTAsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImNvbW1vbmpzIjpmYWxzZSwibm9kZSI6ZmFsc2V9fX0=)

If you'd like the rule to ignore it, you may prefix the variable name with `_`.
This functionality is noted in the docs, as it is built into typescript: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md#ignorednamesregex

---

At the core of your problem is that you'd like the rule to know that removing the argument changes the signature in such a way that "matters" to you. But this removal mattering is an implementation detail of your code, and as such it is impossible for a rule to understand that.
